hypertrace-core-graphql-span-schema/src/main/java/org/hypertrace/core/graphql/span/joiner/SpanJoiner.java

```diff
@@ -12,16 +12,17 @@ public interface SpanJoiner {
   SpanJoiner NO_OP_JOINER =
       new SpanJoiner() {
         @Override
-        public <T> Single<Map<T, Span>> joinSpans(
+        public <T> Single<Map<T, Collection<Span>>> joinSpans(
             Collection<T> joinSources, SpanIdGetter<T> spanIdGetter) {
           return Single.just(Collections.emptyMap());
         }
       };
 
-  <T> Single<Map<T, Span>> joinSpans(Collection<T> joinSources, SpanIdGetter<T> spanIdGetter);
+  <T> Single<Map<T, Collection<Span>>> joinSpans(
```

Comment:
Please return a ListMultimap - we want to maintain order and a map of collections is too ambiguous. I'd also encourage you to leave the existing API so consumers can choose whether they want one or many - it should be trivial to support both since you can just route the singular implementation to the plural with a singleton list.

You'd need to add a couple more interfaces - a `MultipleSpanJoin` and a `MultipleSpanIdGetter` (both currently were made plural but still have singular names).